src/Java.Interop.Tools.Generator/Metadata/NamespaceTransform.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Xml.Linq;
+using Xamarin.Android.Tools;
+
+namespace Java.Interop.Tools.Generator
+{
+	public class NamespaceTransform
+	{
+		public string OldValue { get; }
+		public string NewValue { get; }
+		public bool IsStartsWith { get; }
+		public bool IsEndsWith { get; }
+
+		public NamespaceTransform (string oldValue, string newValue)
+		{
+			OldValue = oldValue;
+			NewValue = newValue;
+
+			if (OldValue.EndsWith (".", StringComparison.Ordinal)) {
+				IsStartsWith = true;
+				OldValue = OldValue.Substring (0, OldValue.Length - 1);
+			}
+
+			if (OldValue.StartsWith (".", StringComparison.Ordinal)) {
+				IsEndsWith = true;
+				OldValue = OldValue.Substring (1);
+			}
+		}
+
+		public string ApplyInternal (string value)
+		{
+			string result;
+
+			while (true) {
+				result = ApplyInternal (value);
+
+				if (result == value)
+					return result;
+
+				value = result;
+			}
+		}
+
+		public string Apply (string value)
+		{
+			// Handle a "starts with" transform
+			if (IsStartsWith) {
+				if (value.StartsWith (OldValue, StringComparison.OrdinalIgnoreCase))
+					return (NewValue + value.Substring (OldValue.Length)).TrimStart ('.');
+
+				return value;
+			}
+
+			// Handle an "ends with" transform
+			if (IsEndsWith) {
+				if (value.EndsWith (OldValue, StringComparison.OrdinalIgnoreCase))
+					return (value.Substring (0, value.Length - OldValue.Length) + NewValue).TrimEnd ('.');
+
+				return value;
+			}
+
+			// Handle an "anywhere" transform
+			var value_tokens = value.Split ('.');
+			var match_tokens = OldValue.Split ('.');
+
+			var results = new List<string> ();
+
+			for (var i = 0; i < value_tokens.Length; i++) {
+				if (AtMatch (value_tokens, i, match_tokens, 0)) {
+					if (NewValue.HasValue ())
+						results.Add (NewValue);
+
+					i += match_tokens.Length - 1;
+				} else {
+					results.Add (value_tokens [i]);
+				}
+			}
+
+			return string.Join (".", results.ToArray ());
```

Comment:
`.ToArray()` shouldn't be needed, as there's a [`string.Join<T>(string?, IEnumerable<T>)` overload](https://docs.microsoft.com/en-us/dotnet/api/system.string.join?view=net-6.0#system-string-join-1(system-string-system-collections-generic-ienumerable((-0)))).